### DISCUSSION TITLE
is nlohmann json recommended to get struct data as json format for low latency application with 1kb data

### DISCUSSION THREAD

**Discussion by naidud2020** [nlohmann/json#3734] [Q&A]
Hi,

Using nlohmann json in low latency application to convert c++ structure data to json message, then send to client.
Here, we found bottleneck with json conversion which is taking 80 Micro seconds, due to this, we are seeing latency now.

is there anyway to reduce 80 to lesser number.

we are doing like below:

void my_struct::get_json(nlohmann::json &j) {
nlohmann json;
json[k1] = this->v1;
:
:
:
json[key_x][sub_key_1]= this->sub_key_val1;
:
:
:
:
}

Any help appreciated.

**Comment by falbrechtskirchinger** [comment]
The strength of this library is convenience and it may not be the library best suited for high-performance or low-latency applications.

That being said, if you're accessing  `json[key_x]` repeatedly, you should obviously cache it.
You could try constructing a `std::map<std::string, nlohmann::json>` directly and move construct a JSON value from it.
You could also try replacing `std::map` with a faster alternative.

Just some ideas, but I suspect you're better off looking for a more performance-/latency-focused library.

**Reply by naidud2020** [reply]
thank you for your quick response. got it.

Below is my use case. is there any alternative way to optimise using nlohmann. already integrated with application.

struct my_struct {
  int v1;
  float v2;
  std::string v3;

   void get_json(nlohmann::json& j)  //**want to convert data to customised json format. _Any suggestions_.**
   {
        j["age"] = this->v1;
        j["sal"]=this->v2;
        :
        :
        json["address"]["street"]= this->v3;
        :
        :
        :
        :
    }
};

Thank you.

**Reply by falbrechtskirchinger** [reply]
I gave you three suggestions for minor improvements. Those are all I got at the moment.

**Reply by naidud2020** [reply]
Okay
